File: src/main/java/com/example/smarthome/SmartHome.java
package com.example.smarthome;

public class SmartHome {

    private final Switch aSwitch;
    private final Switchable switchable;

    public SmartHome(Switch aSwitch, Switchable switchable) {
        this.aSwitch = aSwitch;
        this.switchable = switchable;
    }

    public void run() {
        while (true) {
            if (aSwitch.isOn()) {
                switchable.turnOn();
            } else {
                switchable.turnOff();
            }
        }
    }
}

File: js/src/SmartHome/SmartHome.ts
import {Switch} from './Switch';
import {Switchable} from './Switchable';

export class SmartHome {
  constructor(private readonly aSwitch: Switch,
              private readonly switchable: Switchable) {
  }

  public run(): void {
    if (this.aSwitch.isOn()) {
      this.switchable.turnOn();
    } else {
      this.switchable.turnOff();
    }
  }
}
